src/pages/CustomerInvoicePage.js

```diff
@@ -274,95 +258,91 @@ export class CustomerInvoicePage extends GenericPage {
     }
   }
 
-  renderButtons() {
-    return (
-      <View style={globalStyles.verticalContainer}>
-        <PageButton
-          style={globalStyles.topButton}
-          text={buttonStrings.new_item}
-          onPress={this.openItemSelector}
-          isDisabled={this.props.transaction.isFinalised}
-        />
-        <PageButton
-          text={buttonStrings.add_master_list_items}
-          onPress={this.onAddMasterItems}
-          isDisabled={this.props.transaction.isFinalised}
-        />
-      </View>
-    );
-  }
+  renderButtons = () => (
+    <View style={globalStyles.verticalContainer}>
+      <PageButton
+        style={globalStyles.topButton}
+        text={buttonStrings.new_item}
+        onPress={this.openItemSelector}
+        isDisabled={this.props.transaction.isFinalised}
+      />
+      <PageButton
+        text={buttonStrings.add_master_list_items}
+        onPress={this.onAddMasterItems}
+        isDisabled={this.props.transaction.isFinalised}
+      />
+    </View>
+  );
 
-  render() {
-    return (
-      <GenericPage
-        data={this.state.data}
-        refreshData={this.refreshData}
-        renderCell={this.renderCell}
-        renderTopLeftComponent={this.renderPageInfo}
-        renderTopRightComponent={this.renderButtons}
-        onRowPress={this.onRowPress}
-        onSelectionChange={this.onSelectionChange}
-        onEndEditing={this.onEndEditing}
-        defaultSortKey={this.dataFilters.sortBy}
-        defaultSortDirection={this.dataFilters.isAscending ? 'ascending' : 'descending'}
-        columns={[
-          {
-            key: 'itemCode',
-            width: 2,
-            title: tableStrings.item_code,
-            sortable: true,
-          },
-          {
-            key: 'itemName',
-            width: 4,
-            title: tableStrings.item_name,
-            sortable: true,
-          },
-          {
-            key: 'availableQuantity',
-            width: 2,
-            title: tableStrings.available_stock,
-            sortable: true,
-            alignText: 'right',
-          },
-          {
-            key: 'totalQuantity',
-            width: 2,
-            title: tableStrings.quantity,
-            sortable: true,
-            alignText: 'right',
-          },
-          {
-            key: 'remove',
-            width: 1,
-            title: tableStrings.remove,
-            alignText: 'center',
-          },
-        ]}
-        dataTypesSynchronised={DATA_TYPES_SYNCHRONISED}
-        finalisableDataType={'Transaction'}
-        database={this.props.database}
-        selection={this.state.selection}
-        {...this.props.genericTablePageStyles}
-        topRoute={this.props.topRoute}
+  render = () => (
```

Comment:
Pretty sure this is unnecessary. React.Component lifecycle and render functions don't need any manual binding anyway.

Though I don't see how this'll hurt - unless class variable syntax doesn't play well with inherited methods from React.Component.

Reply:
refactored